### PR TITLE
Qcm

### PR DESCRIPTION
**Pull request by blemia51**

### CODE REVIEW

File: quizz/src/components/ButtonBoolH.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from "react"
```
Comment (ghost) [inline quizz/src/components/ButtonBoolH.js]: If file is obsolete, remove it

File: quizz/src/components/ButtonMask.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (ghost) [inline quizz/src/components/ButtonMask.js:1]: obsolete

File: quizz/src/components/ButtonQcm.css [hunk 3]
```diff
@@ -4,4 +4,79 @@
...
+.noImage{
+    display: none;
+}
+.button-order {
+    display: flex;
+    flex-wrap: wrap;
+    align-items: center;
+    justify-content: center;
+}
+/* .button-order :nth-child(1) {
```
Comment (ghost) [inline quizz/src/components/ButtonQcm.css]: dead code

File: quizz/src/components/ButtonQcm.js [hunk 4]
```diff
@@ -1,34 +1,77 @@
...
+            })
+        }
+
+        componentDidMount() {
+            this.buttonWinOrder();
+            } 
+
     render() {
-    return (
+        console.log(this.state.buttonWinOrder)
```
Comment (ghost) [inline quizz/src/components/ButtonQcm.js]: no more console.log at this state

File: quizz/src/components/ButtonQcm.js [hunk 5]
```diff
@@ -1,34 +1,77 @@
...
-            <button className='qcmButton'>{this.state.incorrect_answer[1]}</button>
-            <button className='qcmButton'>{this.state.incorrect_answer[2]}</button>
-        </div>
-
-    
+  
+        <div className= 'button-order'>
+            <button
+                onClick={event =>{this.setState({win: true});this.props.incrementScore();}}
+                // id='winButton'
```
Comment (ghost) [inline quizz/src/components/ButtonQcm.js]: dead code

File: quizz/src/components/ButtonQcm.js [hunk 6]
```diff
@@ -1,34 +1,77 @@
...
-        <div>
-            <button className='qcmButton'>{this.state.incorrect_answer[1]}</button>
-            <button className='qcmButton'>{this.state.incorrect_answer[2]}</button>
-        </div>
-
-    
+  
+        <div className= 'button-order'>
+            <button
+                onClick={event =>{this.setState({win: true});this.props.incrementScore();}}
```
Comment (ghost) [inline quizz/src/components/ButtonQcm.js]: Send this into a method (handleCorrectAnswer ?)

File: quizz/src/components/ButtonQcm.js [hunk 7]
```diff
@@ -1,34 +1,77 @@
...
+                // id='winButton'
+                style={{ order: this.state.buttonWinOrder }}
+                className={this.state.win||this.state.lost?'winButton':'qcmButton'}
+            >
+                {this.props.correct_answer}
+            </button>
+
+            <button onClick={event =>this.setState({lost: true})}
```
Comment (ghost) [inline quizz/src/components/ButtonQcm.js]: function in onClick is used several times -> store it in a method so you don't repeat yourself (dry)

File: quizz/src/components/ButtonQcm.js [hunk 8]
```diff
@@ -1,34 +1,77 @@
...
+            
+                className={this.state.lost?'lostButton':'qcmButton'} >{this.props.incorrect_answer[2]}
+            </button>
+
+        </div> 
+
+        <div className={this.state.win?'imageWin':'noImage'} >
+             <img className='winLogo' src={"https://i.ibb.co/bFwKdvc/gagne.png"} alt="Win" ></img>
+             <div><button onClick={event =>{this.setState({win: false, lost :false});{this.props.getQuestions()};{this.buttonWinOrder()};{this.props.incrementQuestionNumber()}}}
```
Comment (ghost) [inline quizz/src/components/ButtonQcm.js]: Will change

File: quizz/src/components/CardQuestion.js [hunk 9]
```diff
@@ -1,38 +1,81 @@
...
+    }  
+
+    incrementScore() {
+        this.setState({
+            count: this.state.count + 10
+        })
+    }  
+   
     render() {
+        console.log(this.state.arrayQuestions)
```
Comment (ghost) [inline quizz/src/components/CardQuestion.js]: no more console.log()

File: quizz/src/components/CardQuestionBool.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    }  
+
+    incrementScore() {
+        this.setState({
+            count: this.state.count + 10
+        })
+    }  
+   
+    render() {
+        console.log(this.state.arrayQuestions)
```
Comment (ghost) [inline quizz/src/components/CardQuestionBool.js]: same thing with console.log

File: quizz/src/components/CardQuestionBoolH.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import React from "react"
```
Comment (ghost) [inline quizz/src/components/CardQuestionBoolH.js]: Will be deleted

File: quizz/src/components/ChoiceQuizz.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
```
Comment (ghost) [inline quizz/src/components/ChoiceQuizz.js]: will be deleted

File: quizz/src/components/ChoixQfmVF.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
+import React from "react"
+import './CardQuestion.css'
+import './ButtonQcm.css'
+import './ChoixQfmVF.css'
+import Category from './Category'
+import { Link } from 'react-router-dom'
+
+class ChoixQfmVF extends React.Component {
+    constructor(props) {
```
Comment (ghost) [inline quizz/src/components/ChoixQfmVF.js:9]: don't keep useless constructor

File: quizz/src/components/ChoixQfmVF.js [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        super(props)
+    }
+    
+  render() {
+      console.log(this.props)
```
Comment (ghost) [inline quizz/src/components/ChoixQfmVF.js:14]: conosle.log()

File: quizz/src/components/ListCategory.js [hunk 15]
```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import Category from './Category'
+import './Category.css'
+import Nav from './Nav'
+
+const category = [
```
Comment (ghost) [inline quizz/src/components/ListCategory.js:6]: set data in a separate file

File: quizz/src/components/ListCategory.js [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+            
+        ))}</span>
+    
+    </div>
+);
+
+export default ListCategory;
+
+
+//     <ChoixQfmVF name={this.props.name} image={this.props.image} />
```
Comment (ghost) [inline quizz/src/components/ListCategory.js:86]: ?

File: quizz/src/components/LostMask.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (ghost) [inline quizz/src/components/LostMask.js]: will be deleted

File: quizz/src/components/Maskbutton.js [hunk 18]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (ghost) [inline quizz/src/components/Maskbutton.js]: Will be deleted

File: quizz/src/components/WinButtonPos..js [hunk 19]
```diff
@@ -0,0 +1,13 @@
+function RandomButton(max) {
```
Comment (ghost) [inline quizz/src/components/WinButtonPos..js]: will be deleted

File: quizz/src/components/WinMask.js [hunk 20]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (ghost) [inline quizz/src/components/WinMask.js]: will be deleted

File: quizz/src/components/maskbutton.js [hunk 21]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (ghost) [inline quizz/src/components/maskbutton.js]: will be deleted

**Review by ghost — CHANGES_REQUESTED** [review]

File: quizz/src/components/ButtonBoolH.js (hunk 1 above)
Comment (blemia51) [reply quizz/src/components/ButtonBoolH.js]: done

File: quizz/src/components/ButtonMask.js (hunk 2 above)
Comment (blemia51) [reply quizz/src/components/ButtonMask.js:1]: done

File: quizz/src/components/ButtonQcm.css (hunk 3 above)
Comment (blemia51) [reply quizz/src/components/ButtonQcm.css]: done

File: quizz/src/components/ButtonQcm.js (hunk 4 above)
Comment (blemia51) [reply quizz/src/components/ButtonQcm.js]: ok

File: quizz/src/components/ButtonQcm.js (hunk 5 above)
Comment (blemia51) [reply quizz/src/components/ButtonQcm.js]: done

File: quizz/src/components/LostMask.js (hunk 17 above)
Comment (blemia51) [reply quizz/src/components/LostMask.js]: done

File: quizz/src/components/Maskbutton.js (hunk 18 above)
Comment (blemia51) [reply quizz/src/components/Maskbutton.js]: done

File: quizz/src/components/WinButtonPos..js (hunk 19 above)
Comment (blemia51) [reply quizz/src/components/WinButtonPos..js]: done

File: quizz/src/components/WinMask.js (hunk 20 above)
Comment (blemia51) [reply quizz/src/components/WinMask.js]: done

File: quizz/src/components/maskbutton.js (hunk 21 above)
Comment (blemia51) [reply quizz/src/components/maskbutton.js]: done

**Review by ghost — APPROVED** [review]

**Pull request merged**